promotion/src/main/java/org/cryptimeleon/incentive/promotion/reward/NutellaReward.java

```diff
@@ -0,0 +1,73 @@
+package org.cryptimeleon.incentive.promotion.reward;
+
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+import org.cryptimeleon.incentive.crypto.proof.spend.leaf.TokenUpdateLeaf;
+import org.cryptimeleon.incentive.crypto.proof.spend.tree.SpendDeductTree;
+import org.cryptimeleon.math.serialization.Representation;
+import org.cryptimeleon.math.serialization.annotations.ReprUtil;
+import org.cryptimeleon.math.serialization.annotations.Represented;
+import org.cryptimeleon.math.structures.cartesian.Vector;
+
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Get one free nutella for rewardCost points
+ */
+@Value
+@AllArgsConstructor
+public class NutellaReward implements Reward {
+
+    @Represented
+    @NonFinal
+    Integer rewardCost;
+
+    @Represented
+    @NonFinal
+    UUID rewardId;
+
+    @Represented
+    @NonFinal
+    RewardSideEffect rewardSideEffect;
+
+    public NutellaReward(Representation representation) {
+        ReprUtil.deserialize(this, representation);
+    }
+
+    @Override
+    public SpendDeductTree generateRelationTree(Vector<BigInteger> basketPoints) {
+        // newPoints = 1 * oldPoints + (basketPoints - 4)
```

Comment:
where does the 4 come from?
Is the reward cost somewhere hard-coded to 4?

Reply:
```suggestion
```
It used to be hardcoded, the comment is outdated. Thanks for pointing this out!